caikit/runtime/servicers/runtime_info_servicer.py

```diff
@@ -0,0 +1,78 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Have pylint ignore Class XXXX has no YYYY member so that we can use gRPC enums.
+# pylint: disable=E1101
+
+# Standard
+from typing import Dict
+import importlib.metadata
+import sys
+
+# First Party
+import alog
+
+# Local
+from caikit.config import get_config
+from caikit.interfaces.runtime.data_model import RuntimeInfoResponse
+
+log = alog.use_channel("RI-SERVICR-I")
+
+
+class RuntimeInfoServicerImpl:
+    """This class contains the implementation of all of the RPCs that are required to run a
+    service in Model Mesh as a Model-Runtime."""
+
+    def GetRuntimeInfo(self, request, context) -> RuntimeInfoResponse:  # pylint: disable=unused-argument
+        """Get information on versions of libraries and server for GRPC"""
+        return self.get_runtime_info_impl()
+
+    def get_runtime_info_impl(self) -> RuntimeInfoResponse:
+        """Get information on versions of libraries and server from config"""
+        versions = {}
+        for lib in sys.modules:
+            if (
+                get_config().runtime.versioning
+                and get_config().runtime.versioning.sys_modules
+            ):
+                if len(lib.split(".")) == 1:
+                    version = self.try_lib_version(lib)
+                    if version:
+                        versions[lib] = version
+            # just get caikit versions
+            else:
+                if len(lib.split(".")) == 1 and lib.startswith("caikit"):
```

Comment:
You can easily get `caikit`  version information using `caikit.__version__` or `caikit.__version__tuple__` :

```console
>>> caikit.__version__
'0.22.1.dev10+g11bf1ab'
>>> caikit.__version_tuple__
(0, 22, 1, 'dev10', 'g11bf1ab')
```

Reply:
Good to know! Although we want this to capture all caikit libraries so that can include `caikit_nlp` and `caikit_tgis_backend` as well

Reply:
Yeah, I just figured this was helpful, note that both `caikit_nlp` and `caikit_tgis_backend` also have `__version__`

Reply:
`__version__` seems cleaner way.. @dtrifiro can it also get the version if the package is installed via git commit ?

Reply:
Yep, that's the main advantage in using `setuptools_scm`: version is set at build time by using git tags.